### PR TITLE
String casts binary to Date or DateTime

### PR DESCRIPTION
**Pull request by nurugger07**
The more I look at this the less I agree with it. Looking at other Cast modules they are implemented as I would expect. The Integer module is a good example. Given a string as the value it is converted to an integer. Given an integer it returns an integer.

### CODE REVIEW

**Review by tehprofessor — APPROVED** [review]

File: lib/open_api_spex/cast/string.ex [hunk 1]
```diff
@@ -1,60 +1,79 @@
...
-  defp cast_binary(%{value: value, schema: %{pattern: pattern}} = ctx) when not is_nil(pattern) do
-    if Regex.match?(pattern, value) do
-      Cast.success(ctx, :pattern)
+  def cast(ctx) do
+    Cast.error(ctx, {:invalid_type, :string})
+  end
+
+  ## Private functions
+
+  defp apply_validation(%{value: value, schema: %{maxLength: max_length}} = ctx, [:maxLength|fields])
```
Comment (moxley) [inline lib/open_api_spex/cast/string.ex]: Would you mind applying `mix format` to this file?

Comment (moxley) [reply lib/open_api_spex/cast/string.ex]: I like the rewrite of the multi-try validation!

**Review by moxley — APPROVED** [review]

File: lib/open_api_spex/cast/string.ex (hunk 1 above)
Comment (nurugger07) [reply lib/open_api_spex/cast/string.ex]: Unfortunately, it doesn't appear that the formatter is setup for this project. I can add it as a separate PR but I have a feeling that it's going to affect way more that this one file

Comment (moxley) [reply lib/open_api_spex/cast/string.ex]: It's setup in the project: https://github.com/open-api-spex/open_api_spex/blob/master/.formatter.exs

Perhaps your branch is based off an older version of master?

Comment (nurugger07) [reply lib/open_api_spex/cast/string.ex]: Yeah, not in mine. I'll look into it

Comment (nurugger07) [reply lib/open_api_spex/cast/string.ex]: It's not in the fork which is where my branch is based off

Comment (nurugger07) [reply lib/open_api_spex/cast/string.ex]: Formatter may be set up but the setup is incorrect and still doesn’t appear to have been run. Phoenix is being imported but is only a dep in test environment. The format function won't run with it in the list.

**Pull request closed without merging**